### PR TITLE
refactor!: removed etl::angle; replaced with synfig::Angle

### PR DESCRIPTION
**Pull request by rodolforg**
BREAKING CHANGE: Both ETL and synfig API changed:
- remove `etl::angle`
- `synfig::Angle` is the angle class itself, not a simple `typedef` anymore

### CODE REVIEW

**Comment by rodolforg** [comment]
About LGTM warning:
Captura de tela de 2022-06-27 21-43-57
Detailed info is written here: https://lgtm.com/rules/2160040902/

Should we fix it? I think we actually aren't in this reported situation due to particular case of Angle and Angle::tan classes.

**Comment by ice0** [comment]
> Should we fix it? I think we actually aren't in this reported situation due to particular case of Angle and Angle::tan classes.

I think better to fix it.

**Comment by rodolforg** [comment]
> I think better to fix it.

I'm not sure how to do it. 
The intention of highlighted code is to compute the `atan2()` with the parameters passed to `Angle::tan()` constructor.
A static_cast would solve it? After all, `Angle::tan` is derived from `Angle`. We only want to get the value of its private member `v` (the angle in radians).

```cpp
	Angle a;
	a=static_cast<Angle>(Angle::tan(-centered[1],centered[0]).mod());
	a=a+angle;
```

**Comment by rodolforg** [comment]
(rebased)

**Comment by ice0** [comment]
> I'm not sure how to do it. The intention of highlighted code is to compute the `atan2()` with the parameters passed to `Angle::tan()` constructor. A static_cast would solve it? After all, `Angle::tan` is derived from `Angle`. We only want to get the value of its private member `v` (the angle in radians).

Sorry, was busy last days. 

I wanted to suggest this solution, but you found it first:
```
Angle::tan a=Angle::tan(-centered[1],centered[0]).mod();
a=a+angle;
```

File: synfig-core/src/synfig/angle.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 **
 **	\legal
 **	Copyright (c) 2002-2005 Robert B. Quattlebaum Jr., Adrian Bentley
+**  Copyright (c) 2007 Chris Moore
```
[the remark this answers is not in the document]
Comment (rodolforg) [reply synfig-core/src/synfig/angle.h:7]: Done

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**